ngraph/core/src/op/dot.cpp

```diff
@@ -51,14 +51,19 @@ void op::Dot::validate_and_infer_types()
 {
     element::Type result_et;
 
-    NODE_VALIDATION_CHECK(
-        this,
-        element::Type::merge(result_et, get_input_element_type(0), get_input_element_type(1)),
-        "Arguments do not have the same element type (arg0 element type: ",
-        get_input_element_type(0),
-        ", arg1 element type: ",
-        get_input_element_type(1),
-        ").");
+    const element::Type inputElementType0 = get_input_element_type(0);
+    const element::Type inputElementType1 = get_input_element_type(1);
+    if ((inputElementType0 != element::u8) && (inputElementType1 != element::i8))
```

Comment:
What is this?! could you cover MatMul with TypeRelaxed construction? 
Also MatMul won't be inherited from FusedOp soon, so this Dot class won't be used there.
Also you affect v0::Dot which affect opset1::MatMul which affects opset operation behavior that is not specified!!!